pkg/mapper/mapper.go

```diff
@@ -160,6 +184,19 @@ func (m *MetricMapper) InitFromYAMLString(fileContents string) error {
 
 	m.Defaults = n.Defaults
 	m.Mappings = n.Mappings
+	if n.doFSM {
+		var mappings []string
+		for _, mapping := range n.Mappings {
+			if mapping.MatchType == MatchTypeGlob {
+				mappings = append(mappings, mapping.Match)
+			}
+		}
+		n.FSM.TestIfNeedBacktracking(mappings)
```

Comment:
what happens to the value of this? as it is, isn't this a NOOP?

Reply:
I see now, the method also mutates the FSM, I would prefer not to do that – see my comment there.